x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMHealthBlockedSnapshotIT.java

```diff
@@ -0,0 +1,343 @@
+        IndexRequestBuilder[] builders = new IndexRequestBuilder[numdocs];
+        for (int i = 0; i < builders.length; i++) {
+            builders[i] = prepareIndex(idxName).setId(Integer.toString(i)).setSource("field1", "bar " + i);
+        }
+        indexRandom(true, builders);
+        indicesAdmin().refresh(new RefreshRequest(idxName)).actionGet();
+    }
+
+    private void putSnapshotPolicy(
+        String policyName,
+        String snapshotNamePattern,
+        String schedule,
+        String repoId,
+        String indexPattern,
+        TimeValue unhealthyIfNoSnapshotWithin
+    ) {
+        Map<String, Object> snapConfig = new HashMap<>();
+        snapConfig.put("indices", Collections.singletonList(indexPattern));
+        snapConfig.put("ignore_unavailable", false);
+        snapConfig.put("partial", true);
+
+        SnapshotLifecyclePolicy policy = new SnapshotLifecyclePolicy(
+            policyName,
+            snapshotNamePattern,
+            schedule,
+            repoId,
+            snapConfig,
+            SnapshotRetentionConfiguration.EMPTY,
+            unhealthyIfNoSnapshotWithin
+        );
+
+        PutSnapshotLifecycleAction.Request putLifecycle = new PutSnapshotLifecycleAction.Request(
+            TEST_REQUEST_TIMEOUT,
+            TEST_REQUEST_TIMEOUT,
+            policyName,
+            policy
+        );
+        try {
+            client().execute(PutSnapshotLifecycleAction.INSTANCE, putLifecycle).get();
+        } catch (Exception e) {
+            logger.error("failed to create slm policy", e);
+            fail("failed to create policy " + policy + " got: " + e);
+        }
+    }
+
+    private void assertSlmYellowMissingSnapshot(int numUnhealthyPolicy) throws Exception {
+        assertBusy(() -> {
+            GetHealthAction.Request getHealthRequest = new GetHealthAction.Request(true, 1000);
+            GetHealthAction.Response health = admin().cluster().execute(GetHealthAction.INSTANCE, getHealthRequest).get();
+            assertThat(health.getStatus(), equalTo(HealthStatus.YELLOW));
+            HealthIndicatorResult slmIndicator = health.findIndicator(SlmHealthIndicatorService.NAME);
+            assertThat(slmIndicator.status(), equalTo(HealthStatus.YELLOW));
+            assertThat(slmIndicator.impacts().size(), equalTo(1));
+            List<HealthIndicatorImpact> missingSnapshotPolicies = slmIndicator.impacts()
+                .stream()
+                .filter(impact -> SlmHealthIndicatorService.MISSING_SNAPSHOT_IMPACT_ID.equals(impact.id()))
+                .toList();
+            assertThat(missingSnapshotPolicies.size(), equalTo(numUnhealthyPolicy));
+        });
+    }
+
+    private List<String> executePolicies(String node, List<String> policies) throws Exception {
+        List<String> snapshots = new ArrayList<>();
+        for (String policyName : policies) {
+            snapshots.add(executePolicy(node, policyName));
+        }
+        return snapshots;
```

Comment:
You could also have done
```suggestion
        return policies.stream().map(policy -> executePolicy(node, policy)).toList();
```
to make it a bit more compact (and since performance isn't relevant in this case). But it's only a minor detail, so it's fine if you leave it.

Reply:
I think you could use stream if `executePolicy` doesn't throw checked exceptions.